Source/Scene/TimeDynamicImagery.js

```diff
@@ -0,0 +1,292 @@
+            var currentCache = this._tileCache[currentIntervalIndex];
+            for (var t in currentCache) {
+                if (currentCache.hasOwnProperty(t)) {
+                    currentCache[t].request.cancel();
+                }
+            }
+            delete this._tileCache[currentIntervalIndex];
+            this._tilesRequestedForInterval = [];
+
+            this._currentIntervalIndex = index;
+            this._reloadFunction();
+
+            return;
+        }
+
+        var approachingInterval = getApproachingInterval(this);
+        if (defined(approachingInterval)) {
+            // Start loading recent tiles from end of this._tilesRequestedForInterval
+            //  We keep preloading until we hit a throttling limit.
+            var tilesRequested = this._tilesRequestedForInterval;
+            var success = true;
+            while (success) {
+                if (tilesRequested.length === 0) {
+                    break;
+                }
+
+                var tile = tilesRequested.pop();
+                success = addToCache(this, tile, approachingInterval);
+                if (!success) {
+                    tilesRequested.push(tile);
+                }
+            }
+        }
+    };
+
+    function getKey(x, y, level) {
+        return x + '-' + y + '-' + level;
+    }
+
+    function getKeyElements(key) {
+        var s = key.split('-');
+        if (s.length !== 3) {
+            return undefined;
+        }
+
+        return {
+            x : Number(s[0]),
+            y : Number(s[1]),
+            level : Number(s[2])
+        };
+    }
+
+    function getApproachingInterval(that) {
+        var times = that._times;
+        if (!defined(times)) {
+            return undefined;
+        }
+        var clock = that._clock;
+        var time = clock.currentTime;
+        var isAnimating = clock.canAnimate && clock.shouldAnimate;
+        var multiplier = clock.multiplier;
+
+        if (!isAnimating && multiplier !== 0) {
+            return undefined;
+        }
+
+        var seconds;
+        var index = times.indexOf(time);
+        if (index === -1) {
+            return undefined;
+        }
+
+        var interval = times.get(index);
+        if (multiplier > 0) { // animating forward
+            seconds = JulianDate.secondsDifference(interval.stop, time);
+            ++index;
+        } else { //backwards
+            seconds = JulianDate.secondsDifference(interval.start, time); // Will be negative
+            --index;
+        }
+        seconds /= multiplier; // Will always be positive
+
+        // Less than 5 wall time seconds
+        return (index >= 0 && seconds <= 5.0) ? times.get(index) : undefined;
+    }
+
+    function addToCache(that, tile, interval) {
+        var index = that._times.indexOf(interval.start);
+        var tileCache = that._tileCache;
+        if (!defined(tileCache[index])) {
+            tileCache[index] = {};
```

Comment:
We usually store these in locals to prevent the look ups. So:
```javascript
var tileTimeCache = tileCache[index];
if (!defined(tileTimeCache)) {
    tileTimeCache = tileCache[index] = {};
    // ...
```
And use `tileTimeCache` everywhere else.